src/twisted/internet/test/test_unix.py

```diff
@@ -382,6 +385,56 @@ def test_fileDescriptorOverrun(self):
         test_fileDescriptorOverrun.skip = sendmsgSkip
 
 
+    def test_multiFileDescriptorReceivedPerRecvmsg(self):
+        """
+        Verify that _SendmsgMixin handles receiving multiple file descriptors
+        per recvmsg, calling L{IFileDescriptorReceiver.fileDescriptorReceived}
+        once per received file descriptor.
+        """
+        # Strategy:
+        # - Create a UNIX socketpair.
+        # - Associate one end to a FakeReceiver and FakeProtocol.
+        # - Call sendmsg on the other end with two FDs as ancillary data.
+        # - Call doRead in the FakeReceiver.
+        # - Verify results on FakeProtocol.
+
+        @implementer(IFileDescriptorReceiver)
+        class FakeProtocol(object):
+            def __init__(self):
+                self.fds = []
+            def fileDescriptorReceived(self, fd):
+                self.fds.append(fd)
+                close(fd)
+
+        class FakeReceiver(_SendmsgMixin):
+            bufferSize = 1024
+            def __init__(self, skt, proto):
+                self.socket = skt
+                self.protocol = proto
+            def _dataReceived(self, data):
+                pass
```

Comment:
The appropriate thing to do here would be to have this workaround for now, but also include a link to the "add AF_UNIX to adoptStreamConnection" ticket.  I can't find one - the closest is https://twistedmatrix.com/trac/ticket/5573 which is for adoptStreamPort.